src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -1671,17 +1671,17 @@ void func_80AEE7C4(EnRu1* this, PlayState* play) {
         this->actor.gravity = -((kREG(23) * 0.01f) + 1.3f);
         func_80AED57C(this);
         this->action = 28;
-        *unk_370 = 0.0f;
+        *carryTimer = 0.0f;
     } else if (func_80AEE6D0(this, play)) {
         s32 pad;
 
-        *unk_370 = 0.0f;
+        *carryTimer = 0.0f;
     } else {
         player = GET_PLAYER(play);
         if (player->stateFlags2 & PLAYER_STATE2_IDLE_FIDGET) {
-            this->unk_370 += 1.0f;
+            this->carryTimer += 1.0f;
             if (this->action != 32) {
-                if (*unk_370 > 30.0f) {
+                if (*carryTimer > 30.0f) {
                     if (Rand_S16Offset(0, 3) == 0) {
                         f32 frameCount = Animation_GetLastFrame(&gRutoChildSquirmAnim);
```

Comment:
this timer is specifically only updated while link is idle doing nothing. Its used to see if ruto should do her little "squirming" animation when she gets bored while waiting in links arms for too long. 

I would be more specific like `carryIdleTimer`, `carryIdleFidgetTimer`